### DISCUSSION TITLE
What is Microsoft.AspNetCore.App runtime and why is necessary?

### DISCUSSION THREAD

**Discussion by montyclt** [dotnet/runtime#59902] [Q&A] [answered]
When I run `dotnet --list-runtimes` I get two runtimes installed.

```
Microsoft.AspNetCore.App 5.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]  <-- Why is needed?
Microsoft.NETCore.App 5.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

As I know, an ASP.NET Core application is a standard console application that set up a HTTP Server (I suppose that it is achieved using TCP sockets listening for HTTP messages, but I don't know).

If ASP.NET Core application is a console application, why exists a special runtime for ASP.NET Applications? Can I run ASP.NET applications without installing it?

**Comment by davidfowl** [comment, marked as answer]
It the a set of assemblies that version together that make up asp.net. That's all a shared framework is. It also ships in the box with .net so you always have it. Can you build a console app that uses asp.net? Yes, and you do it by adding a framework reference to Microsoft.AspNetCore.App

**Reply by montyclt** [reply]
As I know, these assemblies are included with an ASP.NET Core application when it is built, why does it come with .NET? In any other framework (for example, MonoGame), framework assemblies are included in application built and aren't included with the runtime.

**Reply by davidfowl** [reply]
It's included because it was the primary application type for building .NET Core projects and there was no mechanism back then for getting other sdks. It's possible that it might be optional in the future using the new workload support but no promises

**Reply by montyclt** [reply]
Thanks David, I have one more related question. This means if I build an ASP.NET Core application, but I don't include the ASP.NET Core framework assemblies, the application still running because it use the assemblies included in the runtime?

**Reply by davidfowl** [reply]
You can't build an ASP.NET Core application without including the framework reference.

**Reply by montyclt** [reply]
Sorry misunderstood, I'm not asking about SDK reference in the csproj file, I'm asking about the generated assemblies in project output (dll files).

Are this files ignored when publishing as framework-dependant because it uses the ASP.NET core assemblies of the runtime?

**Reply by davidfowl** [reply]
Yes. When you add the Framework reference, that gets burned into the runtime manifest (runtimeconfig.json) and those files don't get deployed with the application.

**Reply by montyclt** [reply]
Ok David, thank you!
